jaxrs/WFLY-10636_EAP7-1010.adoc

```diff
@@ -0,0 +1,74 @@
+
+When using reactive types with more than a single potential value (so basically streams), RESTEasy will support three different representation on the wire, dependending on the use of the new org.jboss.resteasy.annotations.Stream annotation on resources:
+
+* no @Stream annotation on the resource method: Resteasy will collect every value until the stream is complete, then wrap them into a java.util.List entity and send to the client
+* @Stream(Stream.MODE.GENERAL): This case uses a variant of the SSE format, modified to eliminate some restrictions inherent in SSE; in particular, certain delimiting characters in the data ('\r', '\n', and '\') are escaped so that arbitrary binary data can be transmitted; moreover the client will close, rather than automatically reconnect, at the end of the stream
+* @Stream(Stream.MODE.RAW): In this case each value is written directly to the wire, without any formatting or delimiters, as it becomes available; this is most useful for values that can be cut in pieces, such as strings, bytes, buffers, etc., and then re-concatenated on the client side.
+
+Since general streaming mode and SSE share minor variants of the same wire protocol, they are, modulo the SSE restriction to character data, interchangeable. As a consequence RESTEasy will make it possible for an SSE client to connect to a resource method that returns a Flowable or an Observable, and a FlowableRxInvoker (implementation of RxInvoker using RxJava(2) Flowable), for example, to connect to an SSE resource method.
+
+== Issue Metadata
+
+=== Issue:
+
+* {issue-base-url}/EAP7-1010[EAP7-1010]
+* {issue-base-url}/WFLY-10636[WFLY-10636]
+
+=== Related Issues:
+
+* {issue-base-url}/RESTEASY-1701[RESTEASY-1701]
+* {issue-base-url}/RESTEASY-1798[RESTEASY-1798]
+
+=== Dev Contacts:
+
+* mailto:asoldano@redhat.com[Alessio Soldano]
+* mailto:rsigal@redhat.com[Ron Sigal]
+* mailto:separdau@redhat.com[Stephane Epardaud]
+* mailto:ema@redhat.com[Jim Ma]
+* mailto:rsearls@redhat.com[Rebecca Searls]
+* mailto:weli@redhat.com[Wei Nan Li]
+
+=== QE Contacts:
+
+* mailto:rsvoboda@redhat.com[Rostislav Svoboda]
+* mailto:kanovotn@redhat.com[Katerina Odabasi]
+* mailto:mkopecky@redhat.com[Marek Kopecky]
+
+=== Affected Projects or Components:
+
+* WildFly
+* RESTEasy
+
+== Requirements
+
+* The support for Reactive Streams types, for both client (RxInvoker) and server (AsyncResponseProvider) sides, will be available and enabled by default.
+* The RxJava(2) implementation will live in a separate resteasy-rxjava2 module that will be available by default but disabled (optional) by default; the user will need to set a dependency to the org.jboss.resteasy.resteasy-rxjava2 module to make use of the RxJava(2) integration.
+* The resteasy-rxjava2 integration / module will be initially supported as tech preview.
```

Comment:
Just clarification here, is this whole RFE is going to be TP? Or only resteasy-rxjava2 module? I understand that the features will not work without this module, but there is also Asynchronous filters support which is not related to rxjava.

In case TP is whole rfe, could you please change it to something like 'This whole RFE is TP'?

Reply:
Now I've explicitly said the whole RFE is expected to be tech preview and turned those 2 words into bold ;-)